src/pybind/mgr/volumes/fs/operations/versions/subvolume_base.py

```diff
@@ -93,7 +93,7 @@ def legacy_config_path(self):
 
     @property
     def namespace(self):
-        return "{0}{1}".format(self.vol_spec.fs_namespace, self.subvolname)
+        return f'{self.vol_spec.fs_namespace}_{self.group.groupname}_{self.subvolname}'
```

Comment:
How will existing setups in the field, with older namespaces (i.e. without the group name) get affected ?

Reply:
I had a chat with Milind regarding this. We've 2 options here -

1. Detect old namespace name and continue using it instead of new namespace name.
2. Detect old namespace name, rename the namespace name to the new name and there onwards use the new namepsace name.

Milind too agrees second option is better. Testing this is will be more complicated than usual functional tests we have. Milind suggested to add a test in fs:upgrade subsuite where on the older version of Ceph cluster, a CephFS subvolume is created within a namespace (which will have its name with the old style) but after the cluster is upgraded this namespace is renamed by mgr/vol code to the new name.

Reply:
I have some questions 
- When does the detection of namespace happen for existing subvolumes?
- How do you rename a namespace? change the layout xattr field?
- Upon rename, would existing data from older namespace transparently be available in the new namespace?

Reply:
> I have some questions
> 
>     * When does the detection of namespace happen for existing subvolumes?

It isn't done in mgr/vol code so far. This PR needs to add it.

> 
>     * How do you rename a namespace? change the layout xattr field?

That's my impression on reading the mgr/vol code but I am not sure. I don't know if simple changing the name ox xattrs filed will have any other side-effects.
 
>
>     * Upon rename, would existing data from older namespace transparently be available in the new namespace?

I don't know. This was going to be my question to you and Milind. :)

Reply:
Thinking further ... a `dir` is associated with a namespace i.e. ceph.dir.layout.pool_namespace. However, the namespace is associated with the entire pool. Essentially, that means that the dir name is equivalent to the namespace, since we don't allow same pool to be reused once it is initialized and attached to a CephFS.

And we shouldn't allow same namespace name to be used for different dirs in the same fs instance ? (this needs to be verified) i.e. data from two distinct subvolume groups or regular dirs shouldn't be clubbed together because of same namespace name ?

So, how do we treat namespace naming for dirs outside a subvolumegroup/subvolume hierarchy ?

Reply:
> > I have some questions
> > ```
> > * When does the detection of namespace happen for existing subvolumes?
> > ```
> 
> It isn't done in mgr/vol code so far. This PR needs to add it.

My question is when do you detect this? When subvolume getpath is run or some other operation? If it possible that CSI mounts the subvolume path and then never ever calls into mgr/volumes?

Reply:
> My question is when do you detect this? When subvolume getpath is run or some other operation?

My initial thought was whenever `def namespace` is accessed, it checks if the current namespace name is of old-style and if so, it renames it.

>  then never ever calls into mgr/volumes?

In that case seems like MGR daemon needs to rename the namespace as soon as the upgrade happens.

Reply:
Spoke to @rishabh-d-dave and @mchangir regarding this post cephfs standup and we need to:

a) figure out if renaming a namespace works and what happens to data under the new namespace.
b) What happens if we make this namespace switch and the subvolume has already been mounted and under use